File: application/play/scala/HomeController.scala
package controllers

import javax.inject._
import play.api._
import play.api.mvc._

/**
 * This controller creates an `Action` to handle HTTP requests to the
 * application's home page.
 */
@Singleton
class HomeController @Inject()(config: Configuration, val controllerComponents: ControllerComponents) extends BaseController {

  /**
   * Create an Action to render an HTML page.
   *
   * The configuration in the `routes` file means that this method
   * will be called when the application receives a `GET` request with
   * a path of `/`.
   */
  def index() = Action { implicit request: Request[AnyContent] =>
    if (config.get[String]("demo.username").isEmpty || config.get[String]("demo.password").isEmpty) {
      InternalServerError("not all variables are set")
    } else {
      Ok("Welcome home " + config.get[String]("demo.username"))
    }
  }
}

File: application/play/java/HomeController.java
package controllers;

import com.typesafe.config.Config;
import play.mvc.*;

import javax.inject.Inject;

/**
 * This controller contains an action to handle HTTP requests
 * to the application's home page.
 */
public class HomeController extends Controller {

    private final Config config;

    @Inject
    public HomeController(Config config) {
        this.config = config;
    }

    /**
     * An action that renders an HTML page with a welcome message.
     * The configuration in the <code>routes</code> file means that
     * this method will be called when the application receives a
     * <code>GET</code> request with a path of <code>/</code>.
     */
    public Result index() {
        if (config.getString("demo.username").isEmpty() || config.getString("demo.password").isEmpty()) {
            return Results.internalServerError("not all variables are set");
        } else {
            return ok("Welcome " + config.getString("demo.username"));
        }
    }
}

